nemo/core/classes/modelPT.py

```diff
@@ -1019,3 +1019,19 @@ def __is_model_being_restored() -> bool:
     def __set_model_restore_state(is_being_restored: bool):
         global _MODEL_IS_RESTORED
         _MODEL_IS_RESTORED = is_being_restored
+
+    def update_weights(self, checkpoint_path):
```

Comment:
To clarify: this function loads the weights only without the optimizer state?

Reply:
My question here is do we want to leverage pytorch instead of lightning to do this? Or should we provide a nemo centric method? It might not be a bad idea to consider unifying or simplifying our loading and saving API.

Reply:
Yes, it just loads the weights as it looks like in new NeMo we can not continue the training while need to update the optimizer. Am I right?
PTL currently does not have such a feature to load the weights. If we can find a better solution, it would be great.

Reply:
> It might not be a bad idea to consider unifying or simplifying our loading and saving API.

Yes please. We have way too many ways to save models - ExperimentManager callbacks, PTL trainer methods, .save_to(), and then basic pytorch save.
Same with restore from but even more convoluted - ExperimentManager restore, PTL restore (in Trainer), restore_from, PT load_state_dict - and this can be state dict of PT (if you manually pytorch save) or PTL saved checkpoint..

Lets discuss this offline

Reply:
Seems like it was resolved in Som's review.